CfpExchange/Controllers/CfpController.cs

```diff
@@ -17,379 +13,341 @@
+        }
+        
+        private async Task<string> GetTimezone(double lat, double lng)
+        {
+            // Only in production, saves credits
+            if (_hostingEnvironment.IsProduction())
+            {
+                using (var httpClient = new HttpClient())
+                {
+                    var resultJson = await httpClient.GetStringAsync($"https://atlas.microsoft.com/timezone/byCoordinates/json?subscription-key={_configuration["MapsApiKey"]}&api-version=1.0&query={lat}%2C{lng}");
+                    var result = JsonConvert.DeserializeObject<TimezoneInfo>(resultJson);
+
+                    return result.TimeZones.FirstOrDefault()?.Id ?? string.Empty;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        public async Task<IActionResult> Details(string id)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+                return RedirectToAction("index", "home");
+
+            var selectedCfp = _cfpService.GetCfpBySlug(id);
+
+            if (selectedCfp == null)
+            {
+                // Check it the id happens to be a Guid
+                if (Guid.TryParse(id, out Guid guidId))
+                {
+                    if (guidId != Guid.Empty)
+                        selectedCfp = _cfpService.GetCfpById(guidId);
+                }
+
+                if (selectedCfp == null)
+                    // TODO to error page?
+                    return RedirectToAction("index", "home");
+            }
+
+            if (selectedCfp.DuplicateOfId != null && selectedCfp.DuplicateOfId != Guid.Empty)
+            {
+                var originalCfp = _cfpService.GetCfpById((Guid) selectedCfp.DuplicateOfId);
+                return RedirectToAction("details", "cfp", new { id = originalCfp.Slug });
+            }
+
+            selectedCfp.Views++;
+
+            await _cfpService.SaveChangesAsync();
+
+            return View(selectedCfp);
+        }
+
+        public async Task<IActionResult> SendReportIssue(Issue issue)
+        {
+            if (ModelState.IsValid)
+            {
+                var bodyText = $"An issue was reported for CFP: https://cfp.exchange/cfp/details/{issue.CfpId}"
+                    + Environment.NewLine + $"Issue: {issue.Description}";
+
+                await _emailSender.SendEmailAsync(_configuration["AdminEmailaddress"],
+                    $"{issue.Name} <{issue.EmailAddress}>", "Issue with CFP", bodyText);
+
+                return Ok();
+            }
+
+            // TODO: return invalid model
+            return BadRequest();
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> OutgoingCfpLink(Guid id, string url)
+        {
+            try
+            {
+                var linkedCfp = _cfpService.GetCfpById(id);
+
+                if (linkedCfp != null)
+                {
+                    linkedCfp.ClicksToCfpUrl++;
+
+                    await _cfpService.SaveChangesAsync();
+                }

```

Comment:
Now the .Single is gone and there's a check for null I think we could remove the try ... catch right?

Reply:
Yep! I don't see a reason why not!

Reply:
Removed